### PR TITLE
first OpenLayerMap files: Draft PR. *notes in OpenLayerMap.tsx*

### PR DESCRIPTION
**Pull request by epachols**
Working on integrating an OpenLayer map into the project, converting an example into Typescript along the way.  

Progress notes - 

'OpenLayerMap.tsx' made for 'Map.jsx' in guide.
needs: 
1. create a marker layer
2. finished TS conversion of react components in model, found in notes on OpenLayerMap.tsx
3. to stand up! Thanks for anyone who has a moment to take a look.

### CODE REVIEW

**Comment by riteshsp2000** [comment]
We are migrating from Mapbox to OpenLayerMap right?

**Comment by ivanoats** [comment]
This is a spike, but it is looking good so far for OpenLayers. It has vector layers which might come in handy.

**Comment by epachols** [comment]
This PR is now available for a test drive, it needs an extra 'npm install' if you already have the project running locally, as it uses two new dependencies.

File: src/App.tsx [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
-import Marker from './Marker'
+import MapContainer from './components/MapContainer/MapContainer'
+// import Map from './Map'
+// import Marker from './Marker'
 
 const App: React.FC = () => {
   return (
     <div className="App">
-      <header className="App-header">
+      {/* <header className="App-header">
```
Comment (ivanoats) [inline src/App.tsx:9]: do we need this comment?

File: src/App.tsx [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
+            width: '50vw',
+            margin: '10vh auto',
           }}
           zoom={[6.5]}
           center={[-123.35, 48.41]}
         >
           <Marker />
         </Map>
-      </header>
+      </header> */}
```
Comment (ivanoats) [inline src/App.tsx:22]: leftover comment?

File: src/components/MapContainer/Layers/VectorLayer.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import React, { useContext, useEffect } from 'react'
+import MapContext from '../Map/MapContext'
+import OLVectorLayer from 'ol/layer/Vector'
+import { vector } from '../Source'
+import GeoJSON from 'ol/format/GeoJSON'
+import { get } from 'ol/proj'
+
+// note: both the VectorLayer styleOptions object and the 'source' from line 33 will need to be hoisted to be able to make multiple different vector layers for different data sources.
```
Comment (ivanoats) [inline src/components/MapContainer/Layers/VectorLayer.tsx:8]: this line's too long to read on github 😆  I'm an old school vimmer, 80 chars is ideal, although I know a lot of people are fine with 120 these days!  up to you.

File: src/components/MapContainer/Map/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    const mapObject = new ol.Map(options)
+    mapObject.setTarget(mapRef.current)
+    setMap(mapObject)
+
+    return () => mapObject.setTarget(undefined)
+  }, [zoom, center])
+
+  // zoom change handler
```
Comment (ivanoats) [inline src/components/MapContainer/Map/Map.tsx:32]: do we need these comments?

**Review by ivanoats — COMMENTED** [review]
💯  awesome 👍🏼   thanks for enabling some serious movement on this project. 

I have a couple of very minor nits on comments to resolve, then it's ready to merge.

File: src/App.tsx (hunk 1 above)
Comment (epachols) [reply src/App.tsx:9]: oop! not in the least

File: src/App.tsx (hunk 2 above)
Comment (epachols) [reply src/App.tsx:22]: just a duplicate I missed, thanks! 👍

File: src/components/MapContainer/Map/Map.tsx (hunk 4 above)
Comment (epachols) [reply src/components/MapContainer/Map/Map.tsx]: I removed them for now, I have the dummy application if I need to put them back in. I figured for now we are just setting a singular map centered around the sound and doing several vector layers.

**Comment by epachols** [comment]
Merging and closing, everything looks good.

**Pull request merged**